### ISSUE TITLE
Improve 'raise' to preserve more type information (and play nicely with smart holes)

### ISSUE THREAD

**Issue by brprice** [hackworthltd/primer#55]
This is a placeholder for now. I should come back and fill this in.
There was some discussion on hackworth.dev on 1st july, and there is some discussion in the commit  "some notes on being careful about syn/chk " 1d2c8d1f791aa7f19a8e8d20876db7f53a01deaa (I will copy this into a comment below, so it does not get lost during a rebase)

## Description

This is a feature request.

<*Include a brief description here.*>

## Dependencies

<*Link to other GitHub Issues or PRs that must be completed before work can begin on this feature.*>

## Spec

<*Describe what the feature will do, from a high-level perspective.*>

## Implementation details

<*If there are any particularly tricky implementation bits that are worth discussing or you haven't quite figured out yet, describe the details here.*>

## Not in spec

<*Describe any features that your implementation is explicitly avoiding, that a reasonable person might think should be in spec. For example, if you're adding a new action that operates on variables, but it only works with term variables and not type variables, you might want to mention it here so that the scope of the feature is clear.*>

## Discussion

<*If there's a GitHub Discussions topic where this feature has been/is being discussed, link it here.*>

## Future work

<*Describe ideas or additional features that might be useful once this feature has been implemented. This is a good place to link to other GitHub Issues or PRs that track this future work.*>
